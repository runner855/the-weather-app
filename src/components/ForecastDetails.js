import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, wind, humidity } = forecast;

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details_date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-details__temperature-min">
        Min Temperature:
        {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__temperature-max">
        Max Temperature:
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__humidity">
        Humidity:
        {`${humidity}%`}
      </div>

      <div className="forecast-details__wind__speed">{`${wind.speed}mph`}</div>

      <div className="forecast-details__wind__direction">{wind.direction}</div>
    </div>
  );
};

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
