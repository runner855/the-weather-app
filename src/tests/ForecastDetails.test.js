import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: 70,
    temperature: {
      min: 12,
      max: 22,
    },
    wind: {
      speed: 40,
      direction: "ne",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
