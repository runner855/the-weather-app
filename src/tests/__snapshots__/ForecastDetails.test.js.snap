// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details_date"
    >
      Thu 1st Jan
    </div>
    <div
      class="forecast-details__temperature-min"
    >
      Min Temperature:12°C
    </div>
    <div
      class="forecast-details__temperature-max"
    >
      Max Temperature:22°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity:70%
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind:40mph
    </div>
  </div>
</DocumentFragment>
`;
